pkg/client/interfaces.go

```diff
@@ -107,6 +108,15 @@ func (gc *GenConfig) Validate() error {
 		if !nameRegexp.MatchString(m.SonobuoyConfig.PluginName) {
 			return fmt.Errorf("invalid plugin name %q; name must only include lowercase alphanumeric values '.' or '-'. This is due to Kubernetes requiring various values to follow RFC 1123 for valid subdomain names", m.SonobuoyConfig.PluginName)
 		}
+
+		for key, value := range m.ConfigMap {
+			if strings.HasSuffix(key, "yml") || strings.HasSuffix(key, "yaml") {
```

Comment:
nit: I didnt catch this when we looked at it, but I'd add the dot as well to look for the suffix: `.yml` and `.yaml`